R/get_source_expressions.R

```diff
@@ -124,229 +124,237 @@ get_source_expressions <- function(filename, lines = NULL) {
 lint_parse_error <- function(e, source_expression) {
   # R >= 4.3.0
   if (inherits(e, "parseError")) {
-    msg <- rex::re_substitutes(e$message, rex::rex(" (", except_some_of(")"), ")", end), "")
-    line <- e$lineno
-    column <- e$colno
-    substr(msg, 1L, 1L) <- toupper(substr(msg, 1L, 1L))
-    msg <- paste0(msg, ".")
-
-    if (inherits(e, "invalidMBCS")) {
-      msg <- paste(msg, "Is the encoding correct?")
-    }
-
-    if (column == 0L) {
-      line <- line - 1L
-      column <- nchar(source_expression$lines[[line]])
-    }
-
-    if (line < 1L || line > length(source_expression$lines)) {
-      # Safely handle invalid location info
-      line <- 1L
-      column <- 1L
-    }
-
-    return(
-      Lint(
-        filename = source_expression$filename,
-        line_number = line,
-        column_number = column,
-        type = "error",
-        message = msg,
-        line = source_expression$lines[[line]]
-      )
-    )
+    return(lint_parse_error_r43(e, source_expression))
   }
 
-  message_info <- re_matches(e$message,
-    rex(except_some_of(":"),
+  message_info <- re_matches(
+    e$message,
+    rex(
+      except_some_of(":"),
       ":",
-      capture(name = "line",
-        digits),
+      capture(name = "line", digits),
       ":",
-      capture(name = "column",
-        digits),
+      capture(name = "column", digits),
       ":",
       space,
-      capture(name = "message",
-        anything),
-      "\n")
+      capture(name = "message", anything),
+      "\n"
     )
+  )
+
+  if (!is.na(message_info$line)) {
+    return(lint_parse_error_r42(message_info, source_expression))
+  }
 
   # an error that does not use R_ParseErrorMsg
-  if (is.na(message_info$line)) {
-
-    if (grepl("invalid multibyte character in parser at line", e$message, fixed = TRUE)) {
-      l <- as.integer(re_matches(
-        e$message,
-        rex("invalid multibyte character in parser at line ", capture(name = "line", digits))
-      )$line)
-      # Invalid encoding in source code
-      return(
-        Lint(
-          filename = source_expression$filename,
-          line_number = l,
-          column_number = 1L,
-          type = "error",
-          message = "Invalid multibyte character in parser. Is the encoding correct?",
-          line = source_expression$lines[[l]]
-        )
+  lint_parse_error_nonstandard(e, source_expression)
+}
+
+lint_parse_error_r43 <- function(e, source_expression) {
```

Comment:
the naming is a bit obscure, we should add internal roxygen comments to these helpers with a little explanation

Reply:
will do